src/cljs/rems/administration/change_catalogue_item_form.cljs

```diff
@@ -0,0 +1,140 @@
+(ns rems.administration.change-catalogue-item-form
+  (:require [re-frame.core :as rf]
+            [rems.administration.administration :refer [administration-navigator-container]]
+            [rems.administration.status-flags :as status-flags]
+            [rems.atoms :as atoms :refer [readonly-checkbox document-title]]
+            [rems.collapsible :as collapsible]
+            [rems.dropdown :as dropdown]
+            [rems.flash-message :as flash-message]
+            [rems.table :as table]
+            [rems.text :refer [localize-time text get-localized-title]]
+            [rems.util :refer [navigate! fetch post!]]))
+
+(rf/reg-event-fx
+ ::enter-page
+ (fn [{:keys [db]} [_ catalogue-items]]
+   {:db (-> (dissoc db ::form)
+            (assoc ::catalogue-items catalogue-items))
+    ::fetch-forms nil}))
+
+(rf/reg-sub ::catalogue-items (fn [db _] (::catalogue-items db)))
+(rf/reg-sub ::form (fn [db _] (::form db)))
+(rf/reg-event-db ::set-form (fn [db [_ form]] (assoc-in db [::form] form)))
+
+(rf/reg-event-db
+ ::update-catalogue-item
+ (fn [db [_ old-catalogue-item-id new-catalogue-item-id new-form]]
+   (update db ::catalogue-items (fn [items]
+                                  (for [item items]
+                                    (if (= (:id item) old-catalogue-item-id)
+                                      (assoc item :id new-catalogue-item-id :formid (:form/id new-form) :form-name (:form/title new-form))
+                                      item))))))
+
+
+(rf/reg-event-fx
+ ::change-catalogue-item-form!
```

Comment:
Also here, it seems that the current convention in the code is to not have the exclamation mark in the event name, even when the effect calls a clojure function with side effects (such as post! here).